src/amb_kf.c

```diff
@@ -162,21 +190,100 @@ static void incorporate_scalar_measurement(u32 state_dim, double *h, double R,
   DEBUG_EXIT();
 }
 
+/** Get the weighted sum of squared innovations.
+ * It's a normalized error metric. High is bad.
+ * More precisely, we square the difference between the predicted observations
+ * and the actual observations, divide them by their variances (FPF' + R),
+ * but pretending they're independent, and then sum them.
```

Comment:
Yeah, we use U. To get the diagonal elements of HPH'+R (which it should have been, per your other comment), we only need the diagonal elements of HPH' and the diagonal elements of R. R happens to be diagonal anyways, but for HPH', we need to use U. We aren't doing anything clever here with U and D.